app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, schema = 'http')
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_host(@host, content_path)
+      return content_path if content_path.start_with?('http', 'https', 'ftp', 'ftps')
```

Comment:
I can `content_path` being something like `tftp_server/my_repo`, using `http://` approach would be probably more bulletproof

Reply:
So i'll change it to something like `%r|^([\w\d_\-\+]+)://|` to capture all uri like code in that case, including for example `file://`, but not something like `ldap:foo...`.